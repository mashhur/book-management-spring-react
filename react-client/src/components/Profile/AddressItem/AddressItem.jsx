import PropTypes from "prop-types";

function AddressItem({ address }) {
	return (
		<button
			type="button mt-2"
			className="list-group-item list-group-item-action rounded-3 p-3"
			aria-current="true">
			<div className="d-flex w-100 justify-content-between align-items-center">
				<div className="me-5">
					<h5 className="my-0">{address.customerName}</h5>
					<p className="my-0 ">
						<small>
							{address.address} {address.mstate}
							{"-"}
							{address.pincode}
						</small>
					</p>
				</div>
			</div>
		</button>
	);
}

AddressItem.propTypes = {
	address: PropTypes.object.isRequired,
};

export default AddressItem;
