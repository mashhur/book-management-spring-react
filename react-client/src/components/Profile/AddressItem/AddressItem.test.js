import React from "react";
import { mount, shallow } from "enzyme";
import {
	checkProp,
	findByTestAttr,
} from "../../../tests/testUtils";
import AddressItem from "./AddressItem";

const defaultProps = {
	address: {
		aid: 39,
		customerName: "Qazi Amaan",
		phoneNumber: "7889776055",
		pincode: "121211",
		mstate: "",
		address: "sdasdas",
	},
};

const setup = (props = {}) => {
	const setUpProps = { ...defaultProps, ...props };
	return shallow(<AddressItem {...setUpProps} />);
};

describe("Address item component test casses", () => {
	it("renders without any error", async () => {
		const wrapper = setup();
		const component = await findByTestAttr(
			wrapper,
			"component-address-item"
		);
		expect(component.length).toBe(1);
	});

	it("doesnot throw warning with expected props", async () => {
		checkProp(AddressItem, defaultProps);
	});

	it("should match the default snapshot", () => {
		const wrapper = setup();
		expect(wrapper).toMatchSnapshot();
	});
});
