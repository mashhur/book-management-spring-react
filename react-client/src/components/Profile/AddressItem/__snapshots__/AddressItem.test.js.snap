// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address item component test casses should match the default snapshot 1`] = `ShallowWrapper {}`;
